AssemblerLib/LocalToGlobalIndexMap.h

```diff
@@ -68,6 +68,12 @@ class LocalToGlobalIndexMap
     /// Returns total number of degrees of freedom.
     std::size_t dofSize() const;
 
+    /// Returns total number of local degrees of freedom for DDC.
+    std::size_t dofSizeLocal() const
+    {
+        return _mesh_component_map.getNLocalUnknowns();
+    }
+
     /// Returns total number of global degrees of freedom for DDC.
     std::size_t dofSizeGlobal() const
```

Comment:
When PETSc is on, dofSize() returns the number of the unknowns of the rank that include those for  ghost nodes. While  dofSizeLocal() returns only the number of the unknowns that are belong to this present partition.